src/Microsoft.AspNet.OData.Shared/EdmEntityObject.cs

```diff
@@ -38,6 +40,31 @@ public EdmEntityObject(IEdmEntityTypeReference edmType)
         public EdmEntityObject(IEdmEntityType edmType, bool isNullable)
             : base(edmType, isNullable)
         {
+            TransientInstanceAnnotationContainer = new ODataInstanceAnnotationContainer();
+            PersistentInstanceAnnotationsContainer = new ODataInstanceAnnotationContainer();
+        }
+
+        /// <inheritdoc />
+        public IODataInstanceAnnotationContainer TransientInstanceAnnotationContainer { get; set; }
+
+        /// <inheritdoc />
+        public IODataInstanceAnnotationContainer PersistentInstanceAnnotationsContainer { get; set; }
+
+        /// <summary>
+        /// Method to Add DAta Modification Exception
```

Comment:
>Modification Exception [](start = 31, length = 22)

Data ModificationException is only one example of a transient instance annotation